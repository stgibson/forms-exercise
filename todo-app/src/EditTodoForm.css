.EditTodoForm {
  margin-top: 20px;
}

.EditTodoForm-input {
  margin-left: 5px;
  margin-right: 10px;
}