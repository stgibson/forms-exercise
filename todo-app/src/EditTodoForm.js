import React, { useState } from "react";
import "./EditTodoForm.css";

const EditTodoForm = ({ todo, editTodo }) => {
  const initFormData = { task: todo.task };
  const [formData, setFormData] = useState(initFormData);

  const handleChange = (event) => {
    const { name, value } = event.target;
    setFormData({ ...formData, [name]: value });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    editTodo(todo.id, formData.task);
  };

  return (
    <form className="EditTodoForm" onSubmit={ handleSubmit }>
      <label htmlFor="edit-task">Edit Task:</label>
      <input
        id="edit-task"
        className="EditTodoForm-input"
        name="edit-task"
        type="text"
        onChange={ handleChange }
        value={ formData.task }
      />
      <button type="submit">Edit Todo</button>
    </form>
  );
};

export default EditTodoForm;