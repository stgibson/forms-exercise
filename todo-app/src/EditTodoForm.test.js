import React from "react";
import EditTodoForm from "./EditTodoForm";
import { render } from "@testing-library/react";

it("renders without crashing", () => {
  render(<EditTodoForm todo={ { task: "Test Task" } } />);
});

it("matches snapshot", () => {
  const { asFragment } =
    render(<EditTodoForm todo={ { task: "Test Task" } } />);
  expect(asFragment()).toMatchSnapshot();
});