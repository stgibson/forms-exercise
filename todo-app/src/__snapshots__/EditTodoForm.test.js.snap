// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="EditTodoForm"
  >
    <label
      for="edit-task"
    >
      Edit Task:
    </label>
    <input
      class="EditTodoForm-input"
      id="edit-task"
      name="edit-task"
      type="text"
      value="Test Task"
    />
    <button
      type="submit"
    >
      Edit Todo
    </button>
  </form>
</DocumentFragment>
`;
